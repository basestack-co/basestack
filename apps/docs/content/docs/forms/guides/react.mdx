---
title: NextJS & React
icon: Code
tags: Basestack Forms, NextJS, React Integration, Form submission, Fetch API, Axios, React, Next.js, API integration, JSON response, Web development, Developer tools, "2025"
---

import { Step, Steps } from 'fumadocs-ui/components/steps';

[React](https://react.dev/?ref=basestack.co) is a JavaScript library for building user interfaces, allowing developers to create interactive, dynamic web applications with reusable components. [Next.js](https://nextjs.org/?ref=basestack.co) is a React framework that enables server-side rendering, static site generation, and simplified routing for building fast, scalable web applications.

To get started, youâ€™ll need the following:

- A [Basestack Forms](https://forms.basestack.co) Account
- A React or Next.js project with a Form component

<Steps>

<Step>
### Use this on your React Project to send form data to your endpoint.

Send data using the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API?ref=basestack.co), [Axios](https://axios-http.com/?ref=basestack.co) or Any [HTTP](https://developer.mozilla.org/en-US/docs/Web/HTTP?ref=basestack.co) Client and ensure the mode on the query string is set to `mode=rest` for retrieving the response in JSON format.
</Step>
<Step>
### Copy and paste this into your React component.

Feel free to use the code below to send a test submission.


```tsx copy title="contact.tsx"
import React, { useState } from "react";

const FormComponent = () => {
  // You can also use rect-hook-form or formik for form handling
  // check out the docs for more info https://react-hook-form.com/
  const [name, setName] = useState("");
  const [email, setEmail] = useState("");
  const [message, setMessage] = useState("");

  const onHandleSubmit = (e) => {
    e.preventDefault();
    e.stopPropagation();

    // You can also use axios or any other library to make the request
    // You can also use async await to make the request
    fetch("https://forms.basestack.co/api/v1/s/[KEY]?mode=rest", {
      method: "POST",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ name, email, message }),
    })
      .then((res) => res.json())
      .then((data) => {
        if (data.code === 200) {
          console.log("Form submitted successfully");
        }
      })
      .catch((error) => {
        console.log("Error submitting form", error);
      });
  };

  return (
    <form onSubmit={(e) => onHandleSubmit(e)}>
      <div>
        <label htmlFor="name">Name:</label>
        {/*  <!-- name each of your inputs as you wish --> */}
        <input
          type="text"
          id="name"
          name="name"
          required
          autoComplete="name"
          aria-label="Name"
          placeholder="Enter your name"
          value={name}
          onChange={(e) => setName(e.target.value)}
        />
      </div>

      <div>
        <label htmlFor="email">Email:</label>
        <input
          type="email"
          id="email"
          name="email"
          required
          autoComplete="email"
          aria-label="Email"
          placeholder="Enter your email"
          value={email}
          onChange={(e) => setEmail(e.target.value)}
        />
      </div>

      <div>
        <label htmlFor="message">Message:</label>
        <textarea
          id="message"
          name="message"
          required
          aria-label="Message"
          placeholder="Enter your message"
          value={message}
          onChange={(e) => setMessage(e.target.value)}
        ></textarea>
      </div>

      {/*  <!-- your other form fields go here --> */}

      <button type="submit">Submit</button>
    </form>
  );
};

export default FormComponent;
```
</Step>
</Steps>
