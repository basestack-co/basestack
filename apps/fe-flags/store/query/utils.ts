import {
  MiddlewareAPI,
  isRejectedWithValue,
  Middleware,
} from "@reduxjs/toolkit";

export const baseUrl = `http://localhost:3000/api/v1`;

export const prepareHeaders = (headers: Headers) => {
  // const state = getState() as RootState;
  return headers;
};

export const rtkQueryErrorLogger: Middleware =
  (api: MiddlewareAPI) => (next) => (action) => {
    if (isRejectedWithValue(action)) {
      const {
        payload: { status },
      } = action;
      // Access forbidden
      if (status === 403) {
        console.log("rtkQueryErrorLogger Access forbidden");
      }
    }
    return next(action);
  };
