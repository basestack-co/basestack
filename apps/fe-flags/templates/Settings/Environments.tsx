import React from "react";
import { SettingCard, Table } from "@basestack/design-system";
import { useDispatch } from "react-redux";
import { AppDispatch } from "store";
import { setCreateEnvironmentModalOpen } from "store/slices/modals";
import { CardList, CardListItem } from "./styles";

const mockMore = [
  { text: "Edit", onClick: () => console.log("") },
  { text: "History", onClick: () => console.log("") },
  { text: "Delete", onClick: () => console.log("") },
];

const mockTableData = {
  headers: ["Environment", "Description", "Show toggle"],
  rows: [
    {
      cols: [
        {
          title: "development",
        },
        {
          title: "Toggles for developers only",
        },
        {
          title: "yes",
        },
      ],
      more: mockMore,
    },
    {
      cols: [
        {
          title: "staging",
        },
        {
          title: "Toggles for Qaâ€™s",
        },
        {
          title: "no",
        },
      ],
      more: mockMore,
    },

    {
      cols: [
        {
          title: "closeprod",
        },
        {
          title: "-",
        },
        {
          title: "no",
        },
      ],
      more: mockMore,
    },

    {
      cols: [
        {
          title: "production",
        },
        {
          title: "Toggles for end users",
        },
        {
          title: "yes",
        },
      ],
      more: mockMore,
    },
  ],
};

const Environments = () => {
  const dispatch = useDispatch<AppDispatch>();

  return (
    <CardList>
      <CardListItem>
        <SettingCard
          title="Environments"
          description="Create and edit environments for feature flags and their rules."
          button="Create New Environment"
          onClick={() => dispatch(setCreateEnvironmentModalOpen(true))}
        >
          <Table data={mockTableData} />
        </SettingCard>
      </CardListItem>
    </CardList>
  );
};

export default Environments;
