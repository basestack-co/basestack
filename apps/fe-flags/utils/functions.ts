// Utils
import isEmpty from "lodash.isempty";

export const isBrowser = typeof window !== "undefined";

export const isNavigator = typeof navigator !== "undefined";

/**
 * UTILS OPERATIONS
 */

export const validURL = (str: string) => {
  const pattern = new RegExp(
    "^(https?:\\/\\/)?" + // protocol
      "((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|" + // domain name
      "((\\d{1,3}\\.){3}\\d{1,3}))" + // OR ip (v4) address
      "(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*" + // port and path
      "(\\?[;&a-z\\d%_.~+=-]*)?" + // query string
      "(\\#[-a-z\\d_]*)?$",
    "i"
  );
  return pattern.test(str);
};

export const urlQueryBuilder = (urlParams: any) => {
  const esc = encodeURIComponent;
  return Object.keys(urlParams)
    .filter((x) => !isEmpty(urlParams[x]))
    .map((k) => esc(k) + "=" + esc(urlParams[k]))
    .join("&");
};

export function groupArrayOfObjects(list: Array<any>, key: string) {
  return list.reduce((rv, x) => {
    (rv[x[key]] = rv[x[key]] || []).push(x);
    return rv;
  }, {});
}

export function groupBy(xs, f) {
  return xs.reduce(
    (r, v, i, a, k = f(v)) => ((r[k] || (r[k] = [])).push(v), r),
    {}
  );
}

export function getBaseUrl() {
  if (typeof window !== "undefined") {
    return "";
  }
  // reference for vercel.com
  if (process.env.VERCEL_URL) {
    return `https://${process.env.VERCEL_URL}`;
  }

  // // reference for render.com
  if (process.env.RENDER_INTERNAL_HOSTNAME) {
    return `http://${process.env.RENDER_INTERNAL_HOSTNAME}:${process.env.PORT}`;
  }

  // assume localhost
  return `http://localhost:${process.env.PORT ?? 3000}`;
}
