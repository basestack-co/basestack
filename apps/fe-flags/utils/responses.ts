// MESSAGES

export const unauthorized = {
  code: 401,
  error: true,
  message: "You must be sign in to view the protected content on this page.",
};

export const methodNotAllowed = {
  code: 405,
  error: true,
  message: `The HTTP method is not supported at this route.`,
};

export const somethingWentWrong = "Something went wrong";

export const notAuthorizedActionProject =
  "You are not authorized to do this action in this project";

// ENVIRONMENTS

export const notAuthorizedCreateEnv =
  "You are not authorized to create an environment";

export const notAuthorizedUpdateEnv =
  "You are not authorized to update this environment";

export const notAuthorizedDeleteEnv =
  "You are not authorized to delete this environment";

// FLAGS

export const notAuthorizedCreateFlag =
  "You are not authorized to create an flag";

export const notAuthorizedUpdateFlag =
  "You are not authorized to update this flag";

export const notAuthorizedDeleteFlag =
  "You are not authorized to delete this flag";

// HISTORY

export const notAuthorizedCreateHistory =
  "You are not authorized to create an history";

export const notAuthorizedUpdateHistory =
  "You are not authorized to update this history";
