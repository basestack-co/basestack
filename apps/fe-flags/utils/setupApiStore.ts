import {
  AnyAction,
  combineReducers,
  configureStore,
  EnhancedStore,
  Middleware,
  Reducer,
} from "@reduxjs/toolkit";

export function setupApiStore<
  A extends {
    reducer: Reducer<any, any>;
    reducerPath: string;
    middleware: Middleware;
    util: { resetApiState(): any };
  },
  R extends Record<string, Reducer<any, any>> = Record<never, never>
>(api: A, extraReducers?: R): { api: any; store: EnhancedStore } {
  /*
   * Modified version of RTK Query's helper function:
   * https://github.com/reduxjs/redux-toolkit/blob/master/packages/toolkit/src/query/tests/helpers.tsx
   */
  const getStore = (): EnhancedStore =>
    configureStore({
      reducer: combineReducers({
        [api.reducerPath]: api.reducer,
        ...extraReducers,
      }),
      middleware: (gdm) =>
        gdm({ serializableCheck: false, immutableCheck: false }).concat(
          api.middleware
        ),
    });

  type StoreType = EnhancedStore<
    {
      api: ReturnType<A["reducer"]>;
    } & {
      [K in keyof R]: ReturnType<R[K]>;
    },
    AnyAction,
    ReturnType<typeof getStore> extends EnhancedStore<any, any, infer M>
      ? M
      : never
  >;

  const initialStore = getStore() as StoreType;
  const refObj = {
    api,
    store: initialStore,
  };
  const store = getStore() as StoreType;
  refObj.store = store;

  return refObj;
}
