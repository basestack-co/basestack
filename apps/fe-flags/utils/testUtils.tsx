import React from "react";
import { render, RenderOptions } from "@testing-library/react";
// Store
import { Provider } from "react-redux";
import { store } from "../store";
// Locales
import { IntlProvider } from "react-intl";
import { messages } from "../locales";
// Styles
import { ThemeProvider } from "styled-components";
import theme from "design-system/theme";
// Auth
import { SessionProvider } from "next-auth/react";

interface AllProvidersProps {
  initialState?: {};
}

const AllProviders: React.FC<AllProvidersProps> = ({
  children,
  initialState = {},
}) => {
  return (
    <SessionProvider session={{ expires: null }}>
      <Provider store={{ ...store, ...initialState }}>
        <IntlProvider locale="en" messages={messages.en}>
          <ThemeProvider theme={theme}>{children}</ThemeProvider>
        </IntlProvider>
      </Provider>
    </SessionProvider>
  );
};

const renderWithAllProviders = (
  ui: React.ReactElement,
  initialState?: {},
  options?: Omit<RenderOptions, "wrapper">
) =>
  render(ui, {
    wrapper: (props) => <AllProviders initialState={initialState} {...props} />,
    ...options,
  });

const WithThemeProvider: React.FC = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithTheme = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: WithThemeProvider, ...options });

const WithReduxProvider: React.FC = ({ children }) => (
  <Provider store={store}>{children}</Provider>
);

const renderWithRedux = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: WithReduxProvider, ...options });

export { renderWithAllProviders, renderWithTheme, renderWithRedux };
