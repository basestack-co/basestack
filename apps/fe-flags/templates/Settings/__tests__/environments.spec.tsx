import React from "react";
import { renderWithAllProviders } from "../../../utils/testUtils";
import { cleanup } from "@testing-library/react";
import Environments from "../Environments";

describe("Environments Template tests", () => {
  afterEach(cleanup);

  test("render Environments correctly", () => {
    const { asFragment } = renderWithAllProviders(<Environments />);

    expect(asFragment()).toMatchSnapshot();
  });
});
