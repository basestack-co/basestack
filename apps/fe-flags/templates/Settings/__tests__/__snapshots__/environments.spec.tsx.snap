// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Environments Template tests render Environments correctly 1`] = `
<DocumentFragment>
  .c5 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  margin-bottom: 1.25rem;
  font-family: 'Roboto',sans-serif;
  color: #000000;
}

.c9 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  font-family: 'Roboto',sans-serif;
  color: #757575;
}

.c12 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  font-weight: 400;
  font-family: 'Roboto',sans-serif;
  color: #000000;
}

.c18 {
  font-size: 0.875rem;
  line-height: 1.375rem;
  font-weight: 400;
  margin-right: 0.75rem;
  font-family: 'Roboto',sans-serif;
  color: #757575;
}

.c4 {
  font-size: 1.125rem;
  line-height: 1.625rem;
  font-weight: 500;
  margin-bottom: 0.25rem;
  font-family: 'Roboto',sans-serif;
  color: #000000;
}

.c14 {
  height: 1.5rem;
  width: 1.5rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c15 {
  color: #000000;
  font-size: 1.5rem;
}

.c19 {
  border: none;
  height: 2.25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 0.75rem;
  font-size: 0.875rem;
  font-weight: 500;
  cursor: pointer;
  border-radius: 4px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out, border-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out, border-color 0.1s ease-in-out;
  background-color: #000000;
  color: #FFFFFF;
  margin-left: auto;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c19:hover:not(:active) {
  background-color: #333;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-shadow: rgba(0,0,0,0.1) 0px 1px 3px 0px,rgba(0,0,0,0.06) 0px 1px 2px 0px;
  background-color: #FFFFFF;
  border-radius: 0.375rem;
}

.c16 {
  height: 1px;
  border: none;
  background-color: #EEEEEE;
  width: 100%;
}

.c13 {
  border: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  border-radius: 50%;
  -webkit-transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out, border-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out, border-color 0.1s ease-in-out;
  background-color: transparent;
  color: #000000;
  height: 2rem;
  width: 2rem;
}

.c13 .material-symbols-sharp {
  color: inherit;
}

.c13:hover {
  background-color: #EEEEEE;
}

.c13:active {
  background-color: #E2E2E2;
}

.c6 {
  background-color: #F6F6F6;
  padding: 1.25rem;
  border-radius: 4px;
}

.c7 {
  display: grid;
  grid-template-columns: repeat(3,1fr) auto;
  grid-gap: 1.25rem;
}

.c7:not(:last-child) {
  border-bottom: 1px solid #E2E2E2;
}

.c10 {
  min-width: 2.25rem;
  pointer-events: none;
}

.c8 {
  height: 2.8125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 1.25rem;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem 1.25rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1:not(:last-child) {
  margin-bottom: 1.25rem;
}

<ul
    class="c0"
  >
    <ul
      class="c1"
    >
      <div
        class="c2"
        data-testid="card"
      >
        <div
          class="c3"
        >
          <h3
            class="c4"
            data-testid="setting-title"
          >
            Environments
          </h3>
          <p
            class="c5"
            data-testid="setting-title"
          >
            Create and edit environments for feature flags and their rules.
          </p>
          <div
            class="c6"
            data-testid="table"
          >
            <div
              class="c7"
              data-testid="header"
            >
              <div
                class="c8"
              >
                <p
                  class="c9"
                >
                  Environment
                </p>
              </div>
              <div
                class="c8"
              >
                <p
                  class="c9"
                >
                  Description
                </p>
              </div>
              <div
                class="c8"
              >
                <p
                  class="c9"
                >
                  Show toggle
                </p>
              </div>
              <div
                class="c10"
              />
            </div>
            <div
              class="c7"
              data-testid="row"
            >
              <div
                class="c8"
              >
                <div
                  class="c11"
                >
                  <p
                    class="c12"
                    font-weight="400"
                  >
                    development
                  </p>
                </div>
              </div>
              <div
                class="c8"
              >
                <div
                  class="c11"
                >
                  <p
                    class="c12"
                    font-weight="400"
                  >
                    Toggles for developers only
                  </p>
                </div>
              </div>
              <div
                class="c8"
              >
                <div
                  class="c11"
                >
                  <p
                    class="c12"
                    font-weight="400"
                  >
                    yes
                  </p>
                </div>
              </div>
              <div
                class="c8"
              >
                <button
                  class="c13"
                >
                  <div
                    class="c14"
                    data-testid="icon-container"
                    size="24"
                  >
                    <span
                      class="c15 material-symbols-sharp"
                      color="#000000"
                      data-testid="icon"
                      size="24"
                    >
                      more_horiz
                    </span>
                  </div>
                </button>
              </div>
            </div>
            <div
              class="c7"
              data-testid="row"
            >
              <div
                class="c8"
              >
                <div
                  class="c11"
                >
                  <p
                    class="c12"
                    font-weight="400"
                  >
                    staging
                  </p>
                </div>
              </div>
              <div
                class="c8"
              >
                <div
                  class="c11"
                >
                  <p
                    class="c12"
                    font-weight="400"
                  >
                    Toggles for Qaâ€™s
                  </p>
                </div>
              </div>
              <div
                class="c8"
              >
                <div
                  class="c11"
                >
                  <p
                    class="c12"
                    font-weight="400"
                  >
                    no
                  </p>
                </div>
              </div>
              <div
                class="c8"
              >
                <button
                  class="c13"
                >
                  <div
                    class="c14"
                    data-testid="icon-container"
                    size="24"
                  >
                    <span
                      class="c15 material-symbols-sharp"
                      color="#000000"
                      data-testid="icon"
                      size="24"
                    >
                      more_horiz
                    </span>
                  </div>
                </button>
              </div>
            </div>
            <div
              class="c7"
              data-testid="row"
            >
              <div
                class="c8"
              >
                <div
                  class="c11"
                >
                  <p
                    class="c12"
                    font-weight="400"
                  >
                    closeprod
                  </p>
                </div>
              </div>
              <div
                class="c8"
              >
                <div
                  class="c11"
                >
                  <p
                    class="c12"
                    font-weight="400"
                  >
                    -
                  </p>
                </div>
              </div>
              <div
                class="c8"
              >
                <div
                  class="c11"
                >
                  <p
                    class="c12"
                    font-weight="400"
                  >
                    no
                  </p>
                </div>
              </div>
              <div
                class="c8"
              >
                <button
                  class="c13"
                >
                  <div
                    class="c14"
                    data-testid="icon-container"
                    size="24"
                  >
                    <span
                      class="c15 material-symbols-sharp"
                      color="#000000"
                      data-testid="icon"
                      size="24"
                    >
                      more_horiz
                    </span>
                  </div>
                </button>
              </div>
            </div>
            <div
              class="c7"
              data-testid="row"
            >
              <div
                class="c8"
              >
                <div
                  class="c11"
                >
                  <p
                    class="c12"
                    font-weight="400"
                  >
                    production
                  </p>
                </div>
              </div>
              <div
                class="c8"
              >
                <div
                  class="c11"
                >
                  <p
                    class="c12"
                    font-weight="400"
                  >
                    Toggles for end users
                  </p>
                </div>
              </div>
              <div
                class="c8"
              >
                <div
                  class="c11"
                >
                  <p
                    class="c12"
                    font-weight="400"
                  >
                    yes
                  </p>
                </div>
              </div>
              <div
                class="c8"
              >
                <button
                  class="c13"
                >
                  <div
                    class="c14"
                    data-testid="icon-container"
                    size="24"
                  >
                    <span
                      class="c15 material-symbols-sharp"
                      color="#000000"
                      data-testid="icon"
                      size="24"
                    >
                      more_horiz
                    </span>
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
        <hr
          class="c16"
          data-testid="horizontal-rule"
        />
        <div
          class="c17"
        >
          <p
            class="c18"
            data-testid="setting-title"
          />
          <button
            class="c19"
          >
            Create New Environment
          </button>
        </div>
      </div>
    </ul>
  </ul>
</DocumentFragment>
`;
